src/components/NewBoardForm.js

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+
+import "./NewBoardForm.css";
+
+
+const kInitialFormData = {
+  title: "",
+  owner: "",
+};
+
+const NewBoardForm = ({ handleNewBoardSubmit }) => {
+  const [formData, setFormData] = useState(kInitialFormData);
+  const [showForm, setShowForm] = useState(true)
+  const [invalidForm, setInvalidForm] = useState({ title: true, owner: true })
+
+  const handleFormVisibility = (event) => {
+    setShowForm(!showForm)
+  }
+
+  const handleFormChange = (event) => {
+    const value = event.target.value;
+    const name = event.target.name;
+    setFormData(prev => ({
+      ...prev, [name]: value
+    }));
+    // Check if there is text in field for a valid submission
+    value ? setInvalidForm(prev => ({ ...prev, [name]: false })) : setInvalidForm(prev => ({ ...prev, [name]: true }))
```

Comment:
This would be better expressed with a full `if` condition. Don't use ternaries for logic. They should be used to pick between one of two values. This code is really using the truthiness of `value` to determine which boolean value to use. If we really wanted to write this concisely, we could do something like:
```js
    setInvalidForm(prev => ({ ...prev, [name]: !Boolean(value) }))
```